### ISSUE TITLE
Android App getting crashed in the Android 12  Caused by: java.lang.NoSuchMethodError: No static method encodeBase64String

### ISSUE THREAD

**Issue by krunalshah131987** [Azure/azure-iot-sdk-java#1518]
The application is working fine in the lower than Android 12 OS mobile phone but for the Android 12 os Mobile phone getting crashed, while the app checking for the device connection status with the IoT Hub, the exception printStackTrace is as below,

 Caused by: java.lang.NoSuchMethodError: No static method encodeBase64String([B)Ljava/lang/String; in class Lorg/apache/commons/codec/binary/Base64; or its super classes (declaration of 'org.apache.commons.codec.binary.Base64' appears in /system/framework/org.apache.http.legacy.jar)
        at com.microsoft.azure.sdk.iot.service.auth.IotHubServiceSasToken.buildToken(IotHubServiceSasToken.java:119)
        at com.microsoft.azure.sdk.iot.service.auth.IotHubServiceSasToken.<init>(IotHubServiceSasToken.java:83)
        at com.microsoft.azure.sdk.iot.service.auth.IotHubServiceSasToken.<init>(IotHubServiceSasToken.java:56)
        at com.microsoft.azure.sdk.iot.service.devicetwin.DeviceTwin.getAuthenticationToken(DeviceTwin.java:747)
        at com.microsoft.azure.sdk.iot.service.devicetwin.DeviceTwin.getTwinOperation(DeviceTwin.java:219)
        at com.microsoft.azure.sdk.iot.service.devicetwin.DeviceTwin.getTwin(DeviceTwin.java:211)

# Context

- **OS and version used: "Android 12, Api level 31"
- **Java runtime used: "1.8"
- **SDK version used:  "com.microsoft.azure.sdk.iot:iot-service-client:1.31.0"

**Comment by timtay-microsoft** [comment]
Android applications have some known compatibility issues with the ```org.apache.commons:comons-lang3``` library in recent versions and this service client package takes a dependency on it. Can you try the lower dependency version of commons-lang3 for your project?

```gradle
dependencies {
    implementation ('com.microsoft.azure.sdk.iot:iot-service-client:1.31.0') {
        exclude(group = "com.apache.commons", module = "commons-lang3")
   }

    implementation ('com.apache.commons:commons-lang3:3.6')
}
```

**Comment by timtay-microsoft** [comment]
any luck with the above suggestion @krunalshah131987 ?

**Comment by timtay-microsoft** [comment]
I'm assuming that you are unblocked, so I'm closing this issue. Feel free to open a new issue if you run into any further issues.
